rkt/images.go

```diff
@@ -636,6 +713,11 @@ func (f *fetcher) downloadHTTP(url, label string, out writeSyncer, etag string)
 		return nil, fmt.Errorf("bad HTTP status code: %d", res.StatusCode)
 	}
 
+	err = ioutil.WriteFile(etagfilepath, []byte(res.Header.Get("ETag")), 0644)
+	if err == nil {
+		defer os.Remove(etagfilepath)
```

Comment:
A download can be interrupted for different reasons:
- 1. the user kills rkt or reboots the computer. In this case, it will not be deleted (that's what we want).
- 2. the server is killed or the internet connection gets disconnected: in this case, the copy fails with an error but the `defer` will still remove the file, so the next download will unfortunately start from the beginning.

I would ignore the error if `ioutil.WriteFile` fails, and move the `os.Remove` below `io.Copy` or `out.Sync`.

The tests currently only test the case 1 and not the case 2 btw.